File: rich_game_mechnism/src/com/tw/core/commands/SellTool.java
package com.tw.core.commands;

import com.tw.core.Player;
import com.tw.core.responses.Response;
import com.tw.core.tools.Tool;

/**
 * Created by pzzheng on 11/17/16.
 */
public class SellTool implements Command {
    private Tool tool;

    public SellTool(Tool tool) {
        this.tool = tool;
    }

    @Override
    public Player.Status execute(Player player) {
        if(player.getAsests().hasTool(tool)){
            player.getAsests().removeTool(tool);
            player.getAsests().addPoints(tool.getValue());
        }
        return player.waitForCommand();

    }

    @Override
    public Player.Status respond(Response response, Player player) {
        return null;
    }
}

File: ruby/rich_game_ruby/src/commands/sell_tool.rb
require_relative 'command'
class SellTool < Command
  def initialize(tool)
    @tool = tool
  end

  def execute(player)
    if(player.asset.hasTool(@tool)) then
      player.asset.sellTool(@tool)
    end
    return player.waitForCommand
  end


end